### DISCUSSION TITLE
Type conversion with default values

### DISCUSSION THREAD

**Discussion by DocDriven** [nlohmann/json#3537] [Q&A] [answered]
I am looking for a way to access a value with default value, that allows me to cast them to uint64_t.
I have tried the following code, but it does not seem to do the conversion correctly.
```
int main(int argc, char* argv[])
{
    json j;
    j["val"] = 228128275778072576;

    uint64_t v = j.value("val", 0);
    std::cout << v << std::endl;

    return 0;
}
```
This prints 14680064 instead.
It is possible to combine the cast with the checked and unchecked access functions, but not with value, e.g.
```
j["val"].get<uint64_t>()
j.at("val").get<uint64_t>()
```
both work, but I cannot provide a default value. Is there a reason why this does not work with value, and is there an elegant way to do it?

**Comment by gregmarr** [comment, marked as answer]
Have you tried `auto v = j.value("val", uint64_t(0));` or `auto v = j.value<uint64_t>("val", 0)`?

**Reply by nlohmann** [reply]
@DocDriven Do you need further assistance?
